tqecjs/src/workspace/plaquettes/PlaquetteClass.js

```diff
@@ -216,26 +215,32 @@ export default class Plaquette extends Graphics {
 	onDragStart(event) {
 		this.isDragging = true;
 		this.initialPosition = event.data.getLocalPosition(this.parent);
-		console.log(this.initialPosition);
+		console.log("Initial position: " + this.initialPosition);
 	}
 
 	onDragMove = (event) => {
+		if (!this.isDragging) {
+			return;
+		}
 		let newQubits = [];
 		let diff = 0;
 		// Get current position
 		const newPosition = event.data.getLocalPosition(this.parent);
 		// Calculate the distance moved
 		const deltaX = newPosition.x - this.initialPosition?.x;
 		const deltaY = newPosition.y - this.initialPosition?.y;
-		console.log(deltaY, deltaY);
+		console.log(`Drag (dx, dy) = (${deltaX}, ${deltaY})`);
 		let shiftQ = null;
-		if (
-			this.isDragging &&
-			(deltaX >= this.gridSize * 0.4 || deltaY >= this.gridSize * 0.4)
-		) {
+		const scaleFactor = 0.4;
+		const threshold = this.gridSize * scaleFactor;
+		// If the cursor is moved meaningfully, check which direction to copy to
+		if (Math.abs(deltaX) >= threshold || Math.abs(deltaY) >= threshold) {
```

Comment:
Although there are times when the canvas continues to make plaquettes after the user has stopped cliccking down on the mouse pad and is just dragging their cursor, the new changes do make things smoother